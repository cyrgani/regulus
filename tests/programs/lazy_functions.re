def(ignore, $a, null),
ignore(print("hidden")),

def(use_twice_without_calling, $a, _(
    a, a, null
)),
use_twice_without_calling(print("also hidden")),

def(use_once, $a, a()),
use_once(print("once #1")),

# TODO: This should only cause `a` to be evaluated once, not twice.
def(use_twice, $a, _(
    a(), a(), null
)),
use_twice(print("once #2")),
