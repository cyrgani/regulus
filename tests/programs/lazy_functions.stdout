once #1
once #2
once #2
